spec/test_app_templates/Gemfile.extra

```diff
@@ -7,3 +7,8 @@ else
   # bootstrap 4
   gem "bootstrap"
 end
+
+# Can't explain why these restrictions are necessary for circle-ci to pass, they
+# should not be, and do not seem required to have tests pass locally.
+gem 'puma', "~> 3.11"
+gem 'sprockets', "~> 3.7"
```

Comment:
Not sure if these should be in `Gemfile` instead of `Gemfile.extra`. If `Gemfile.extra` is present, it seems an appropriate place for these... but I also don't understand why it's present, and am worried it will mess up the circleci samvera orb which has rules for cache keys based on checksums of the `Gemfile` but not necessarily `Gemfile.extra`. 

I would be inclined to get rid of this `Gemfile.extra` entirely and just put all of these in the `Gemfile`. But I may not have the context of why `Gemfile.extra` was being used (and apparently recommended by `engine_cart`?) in the first place... which also means I may not be making the right decision about when to put things there.